### PR TITLE
Data api - Urgency Detection Queries

### PR DESCRIPTION
**Pull request by sidravi1**
Reviewer: @suzinyou 
Estimate: 10 mins

---

## Ticket

Fixes: [AAQ-675](https://idinsight.atlassian.net/browse/AAQ-675)

### Goal

As as previous PR but adds the Urgency Detection History

### Changes
- New endpoint
- New tests
 
## How has this been tested?
New tests

## Checklist

Fill with `x` for completed. 

- [x] My code follows the style guidelines of this project
- [x] I have reviewed my own code to ensure good quality
- [x] I have tested the functionality of my code to ensure it works as intended
- [x] I have resolved merge conflicts

(Delete any items below that are not relevant)
- [x] I have updated the automated tests
- [ ] I have updated the scripts in `scripts/`
- [ ] I have updated the requirements
- [ ] I have updated the README file
- [ ] I have updated affected documentation
- [ ] I have added a blogpost in Latest Updates
- [ ] I have updated the CI/CD scripts in `.github/workflows/`
- [ ] I have updated the Terraform code

[AAQ-675]: https://idinsight.atlassian.net/browse/AAQ-675?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: core_backend/app/data_api/routers.py [hunk 1]
```diff
@@ -87,43 +157,19 @@ def convert_to_pydantic_model(query: QueryDB) -> QueryExtract:
...
-            QueryResponseErrorExtract(
-                error_id=query.response_error[0].error_id,
-                error_message=query.response_error[0].error_message,
-                error_type=query.response_error[0].error_type,
-                error_datetime_utc=query.response_error[0].error_datetime_utc,
-            )
-            if query.response_error
-            else None
-        ),
+        response=[
```
Comment (sidravi1) [inline core_backend/app/data_api/routers.py:160]: Realised i could just use model_validate! Much smaller

Comment (suzinyou) [reply core_backend/app/data_api/routers.py:160]: Oh true!!

File: pyproject.toml [hunk 2]
```diff
@@ -12,6 +12,12 @@ line-length = 88
 [tool.ruff.lint.flake8-bugbear]
 extend-immutable-calls = ["fastapi.Depends", "fastapi.params.Depends","typer.Option"]
 
+[tool.ruff.lint.isort]
+known-first-party = ["corebackend"]
+
+[tool.isort]
+known_first_party = ["corebackend"]
```
Comment (suzinyou) [inline pyproject.toml:19]: What do these do?

File: core_backend/tests/api/test_data_api.py [hunk 3]
```diff
@@ -37,6 +43,145 @@ def utcnow(self) -> datetime:
...
+class TestUrgencyQueryDataAPI:
+    @pytest.fixture
+    async def user1_data(
+        self,
+        monkeypatch: pytest.MonkeyPatch,
+        asession: AsyncSession,
+        urgency_rules: int,
+    ) -> AsyncGenerator[None, None]:
+        dates = [
+            datetime.utcnow() - relativedelta(days=x) for x in range(N_DAYS_HISTORY)
```
Comment (suzinyou) [inline core_backend/tests/api/test_data_api.py:55]: I get this warning on all `datetime.utcnow()`! :

> The method "utcnow" in class "datetime" is deprecated
>  Use timezone-aware objects to represent datetimes in UTC; e.g. by calling .now(datetime.UTC)

File: core_backend/tests/api/test_data_api.py [hunk 4]
```diff
@@ -37,6 +43,145 @@ def utcnow(self) -> datetime:
...
+
+        response = client.get(
+            "/data-api/urgency-queries",
+            headers={"Authorization": "Bearer test_api_key"},
+            params={"start_date": "2021-01-01T00:00", "end_date": "2021-01-10T00:00"},
+        )
+        assert response.status_code == 200
+
+    @pytest.mark.parametrize(
+        "start_day, end_day", [[5, 1], [6, 4], [5, 5], [0, 0], [20, 14], [11, 0]]
```
Comment (suzinyou) [inline core_backend/tests/api/test_data_api.py:122]: nit: the names start_day and end_day are a bit confusing unless we look at how these values are used.. but unsure what that alternative naming can be!

Comment (suzinyou) [inline core_backend/tests/api/test_data_api.py:122]: Also, it seems like we are allowing start_date > end_date in query params (just returns empty list. If this is intentional should we add such test cases too? e.g. [1, 5], [4, 6], ...

Comment (suzinyou) [reply core_backend/tests/api/test_data_api.py:122]: Oh I see you used `days_ago` in the fixture! maybe `days_ago_start` `days_ago_end`? But again, optional!

**Review by suzinyou — APPROVED** [review]
Thank you, I've tested with front end and swagger UI!

Just a few small comments!

File: core_backend/tests/api/test_data_api.py (hunk 3 above)
Comment (sidravi1) [reply core_backend/tests/api/test_data_api.py:55]: Yep! I'll update it. I'm rebasing over Amir's change now.

File: core_backend/tests/api/test_data_api.py (hunk 4 above)
Comment (sidravi1) [reply core_backend/tests/api/test_data_api.py:122]: Thanks!

Comment (sidravi1) [reply core_backend/tests/api/test_data_api.py:122]: Good idea. I'll add that

File: pyproject.toml (hunk 2 above)
Comment (sidravi1) [reply pyproject.toml:19]: Tells isort that "corebackend" is our library so it's sorts it correctly. Usually it does a good job by itself but was failing for me locally for some reason.

https://pycqa.github.io/isort/docs/configuration/options.html#known-third-party

Comment (suzinyou) [reply pyproject.toml:19]: shouold it have been `core_backend`?

Comment (sidravi1) [reply pyproject.toml:19]: ugh. Yeah :/ I'll fix in. the next one

**Pull request merged**